docs/DeveloperGuide.md

```diff
@@ -1,38 +1,287 @@
+- `Ui`: In charge of printing messages
+- `Parser`: Determines the command to execute
+- `Modes`: Determines the group of functionalities that the user wants to use.
+- `FileManager`: Read and write data from hard disk
+
+### Program Flow
+The sequence diagram below describes how the components interacts with one another when the user issues a command.
+
+For simplicity, the 3 different modes will be classified as `:Mode`.
+
+<u>Sequence Diagram for PlanPal</u>  
+![MainProgramFlowDiagram.drawio.png](Images%28DG%29%2FMainProgramFlowDiagram.drawio.png)
+
+
+## Ui
+
+<u>Overview</u>  
+The `Ui` component handles operations that displays messages for the user to read as an instruction.  
+Additionally, it is also used to capture user inputs in certain functions.
+
+<u>Methods</u>  
+The `Ui` component has the following key methods:
+
+| Method                | Description                                                          |
+|-----------------------|----------------------------------------------------------------------|
+| printWelcomeMessage   | Prints the welcome message                                           |
+| printAvailableModes   | Print all modes that are currently available in PlanPal              |
+| printByeMessage       | Prints the good bye message                                          |
+| printCreateStorage    | Prints the path to the storage that is created                       |
+| getSetCategory        | Capture user input when using the category function in certain modes |
+| printCat              | Prints the categories available in certain modes                     |
+| printCategoryNotFound | Prints when category is not found                                    |
+| print                 | Custom print function that accepts multiple strings                  |
+
+<u>Design Considerations</u>  
+This component was created as functionalities such as printing will be used in most of the classes in the PlanPal Application.  
+Consolidating all the print methods that are repeated reduces repetition in code and makes it easier to update when necessary.
+
+## Parser
+<u>Overview</u>
+This component handles the logic behind the application. The parser component consists of the parent `Parser` Class and 3 children that inherits the `Parser` Class. These 3 children are used when the respective modes are in play.  
+
+<u>Class Diagram for Parser Component</u>  
+![ParserClassDiagram.drawio.png](Images%28DG%29%2FParserClassDiagram.drawio.png)
+
+<u>Design Considerations</u>  
+The Parser class follows this structure as there are common commands between the different modes. Using inheritance prevents the repetition of code. Additionally, different parsers were created since different modes require different functionalities.
+
+## Mode: Contact Manager
+The class diagram below represents the contact book system
+
+<u>Class Diagram for Contact Manager</u>  
```

Comment:
Is there a reason why different class diagrams look like they are generated from different software?

Also, do have the association labels, roles, and multiplicities.

Otherwise, actually a pretty good class diagram that shows the overview of the ContactManager!